feature-libs/order/core/facade/myaccount-v2-order-history.service.ts

```diff
@@ -0,0 +1,268 @@
+        const orderView: OrderView = { ...order };
+        orderView.consignments = [];
+        const requests = (order.consignments ?? []).map(
+          (consignment: Consignment) => {
+            const consignmentView: ConsignmentView = { ...consignment };
+            if (consignment.code && consignment.trackingID) {
+              return this.getConsignmentTracking(
+                order?.code ?? '',
+                consignment.code
+              ).pipe(
+                map((trackingInfo: ConsignmentTracking) => {
+                  consignmentView.consignmentTracking = trackingInfo;
+                  orderView.consignments?.push(consignmentView);
+                  return orderView;
+                })
+              );
+            } else {
+              orderView.consignments?.push(consignmentView);
+              return of(orderView);
+            }
+          }
+        );
+        if (requests === undefined || requests.length < 1) {
+          return of(orderView);
+        }
+        return combineLatest(requests).pipe(
+          switchMap((orders: OrderView[] | undefined) => {
+            if (orders !== undefined) {
+              return of(orders[0]);
+            } else {
+              return of(order);
+            }
+          })
+        );
+        //<-----------------
+      })
+    );
+  }
+
+  getOrderHistoryListWithDetails(
+    pageSize: number
+  ): Observable<OrderHistoryListView | undefined> {
+    const orderListView: OrderHistoryListView = {};
+    return this.orderHistoryService.getOrderHistoryList(pageSize).pipe(
+      switchMap((orderList: OrderHistoryList | undefined) => {
+        orderListView.pagination = orderList?.pagination;
+        orderListView.sorts = orderList?.sorts;
+        orderListView.orders = [];
+        const requests = (orderList?.orders ?? []).map(
+          (order: OrderHistory) => {
+            const orderView: OrderHistoryView = { ...order };
+            return this.getOrderDetailsWithTracking(order?.code ?? '').pipe(
+              map((orderDetail: OrderView | undefined) => {
+                /** filling extra fields ---> */
+                orderView.entries = orderDetail?.entries;
+                orderView.consignments = orderDetail?.consignments;
+                orderView.unconsignedEntries = orderDetail?.unconsignedEntries;
+                orderView.returnRequests = [];
+                /** filling extra fields <--- */
+                orderListView.orders?.push(orderView);
+                return orderListView;
+              })
+            );
+          }
+        );
+        if (requests.length === 0) {
+          // in case of no order
+          requests.push(of(orderListView));
+        }
+        return combineLatest(requests);
+      }),
+      map((requests: OrderHistoryListView[] | undefined) => {
+        if (requests !== undefined) {
+          return requests[0];
+        } else {
+          return {};
+        }
+      })
+    );
+  }
```

Comment:
[MINOR]
As we discussed offline, it would be good to simplify the implementation of this method to:
- get rid of a shared static variable `orderListView` and use pure reactive programming
- return `Observable<OrderView>` instead of `OrderHistoryListView` from `map()` in the `requests`. In other words the `const requests` should have type `Observable<OrderView>[]` 
- then combine the observables and create a `OrderHistoryView`. Curently we do a non-intuitive trick of returning `requests[0];` (becasue our `requests` array is of type `Observable<OrderViewList>[]`

Note: the implementation of this method can be refactored to be much simpler to rason about, BUT this is not part of our public API, so can be fixed even after the release.